### ISSUE TITLE
Stepwise regression doesn't use test/train validation when producing intermediate scores

### ISSUE THREAD

**Issue by berkgercek** [berkgercek/neurencoding#4]
The current implementation of stepwise regression via [`neurencoding.utils.SequentialSelector`](https://github.com/berkgercek/neurencoding/blob/ffa1b3d4d810e3a12905d2cf10ba509fedd99e18/neurencoding/utils.py#L101) does not use the test data when computing scores at each added regressor.

This is because if the model uses the testing data to select a new regressor during the sequential process this is technically a violation of the test/train paradigm. As such only the improvement on the training data is reported in the `.deltas_` and `.scores_` attributes. 

The real solution is to use the testing data to select the next regressor and store the improvement for the new model on *both* the testing and training data at each iteration. This needs to be done.

**Comment by berkgercek** [comment]
Okay so with [this commit](https://github.com/berkgercek/neurencoding/commit/718e00fde0299c957338fc729b4b159ced486d2f) it should be fully functional using the SequentialSelector class to perform stepwise regression. 

The next feature is chosen by scoring on the train dataset, but the score for the same feature on the test set is also provided and stored. That way we can have a full coverage of the data and report test scores. New attributes of sfs are:

```
sfs = SequentialSelector(nglm)
sfs.fit(full_scores=True)
sfs.scores_test_
sfs.scores_train_
sfs.deltas_test_
sfs.deltas_train_
# optionally, if the full_scores flag is passed as true:
sfs.full_scores_test_
sfs.full_scores_train_
```
The `full_scores` option is to store all scores for the intermediate candidate features, not just the maximum.
